internal/fields/validate.go

```diff
@@ -490,3 +497,12 @@ func ensureConstantKeywordValueMatches(key, value, constantKeywordValue string)
 	}
 	return nil
 }
+
+// ensureAllowedValues validates that the document's field value
+// is one of the allowed values.
+func ensureAllowedValues(key, value string, allowedValues AllowedValues) error {
```

Comment:
nit: in theory, you don't need to pass `key` here as it's just used to render the error message :)

Reply:
Umm, but error message is rendered here, `key` is used in this function for that.

I would be fine with refactoring this error generation, but in a different PR, making the other ensure method consistent with this.